src/core/stream/representation/types.ts

```diff
@@ -20,6 +17,7 @@ import type { IRange } from "../../../utils/ranges";
 import type { IReadOnlySharedReference } from "../../../utils/reference";
 import type { SegmentQueue } from "../../fetchers";
 import type { IBufferType, SegmentSink } from "../../segment_sinks";
+import type { IAdaptationStreamPlaybackObservation } from "../adaptation";
```

Comment:
I wrote it as a comment but you may have not seen it:

> As the AdaptationStream is the one that has a dependency on the RepresentationStream, shouldn't the type be coming from the RepresentationStream to the AdaptationStream and not the reverse?

Reply:
`IAdaptationStreamPlaybackObservation` and `IRepresentationStreamPlaybackObservation` are the same so I deleted the Representation one.
Yeah name is a bit off because of that, should I rename it? Any idea of a name that would fit for both? Or I can duplicate it but I'm not fan of..